File: 96.不同的二叉搜索树.py
#
# @lc app=leetcode.cn id=96 lang=python
#
# [96] 不同的二叉搜索树
#

# @lc code=start
class Solution(object):
    def numTrees(self, n):
        """
        :type n: int
        :rtype: int
        """
        store= [1, 1]
        if n <= 1:
            return store[n]
        for m in range(2, n+1):
            s = m-1
            count = 0
            for i in range(m):
                count += store[i]*store[s-i]
            store.append(count)
        return store[n]
# @lc code=end

File: cpprounttmp/96.不同的二叉搜索树.cpp
/*
 * @lc app=leetcode.cn id=96 lang=cpp
 *
 * [96] 不同的二叉搜索树
 */

// @lc code=start
class Solution {
public:
    int numTrees(int n) {
        if(n < 3)
            return n;
        return  n + numTrees(n-1);

    }
};
// @lc code=end

